cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -181,8 +179,7 @@ public void setAuditPublisher(AuditPublisher auditPublisher) {
   public void dropAllInNamespace(final NamespaceId namespace) throws Exception {
     // To delete all streams in a namespace, one should have admin privileges on that namespace
     ensureAccess(namespace, Action.ADMIN);
-    UserGroupInformation ugi = impersonator.getUGI(namespace);
-    Iterable<Location> locations = ImpersonationUtils.doAs(ugi, new Callable<Iterable<Location>>() {
+    Iterable<Location> locations = impersonator.doAs(namespace, new Callable<Iterable<Location>>() {
```

Comment:
I answered the ADMIN part of the question in the linked comment. In our current behavior namespace owner had 'RWXCA' on the db so he was able to delete the table. We need more testing with HBase authorization.